jwst/lib/set_telescope_pointing.py

```diff
@@ -1364,6 +1395,36 @@ def all_pointings(mnemonics):
     return pointings
 
 
+
+def populate_model_from_siaf(model, siaf):
+    """
+    Populate the WCS keywords of a Level1bModel from the SIAF.
+
+    Parameters
+    ----------
+    model : `~jwst.datamodels.Level1bModel`
+        Input data as Level1bModel.
+    siaf : namedtuple
+        The WCS keywords read in from the SIAF.
+    """
+    # If not an imaging mode, update the pointing only.
+    model.meta.wcsinfo.v2_ref = siaf.v2_ref
+    model.meta.wcsinfo.v3_ref = siaf.v3_ref
+    model.meta.wcsinfo.v3yangle = siaf.v3yangle
+    model.meta.wcsinfo.vparity = siaf.vparity
+    if model.meta.exposure.type.lower() in TYPES_TO_UPDATE:
```

Comment:
Again, is this OK for the guide star mode data? The guide modes don't have aperture entries in the SIAF and hence siaf.crpix1 and siaf.crpix2 won't have values that can be extracted directly from the SIAF. They require alternate computation (I believe we simply set them to the middle of the image data array?).